assets/js/components/page-header.js

```diff
@@ -21,98 +21,69 @@
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
 
-/**
- * WordPress dependencies
- */
-import { Component } from '@wordpress/element';
-import { applyFilters } from '@wordpress/hooks';
-
-/**
- * Internal dependencies
- */
-import SvgIcon from '../util/svg-icon';
-import DateRangeSelector from './date-range-selector';
-
-class PageHeader extends Component {
-	render() {
-		const { title, icon, iconWidth, iconHeight, iconID, className, status, statusText, fullWidth } = this.props;
+function PageHeader( props ) {
+	const { title, icon, className, status, statusText, fullWidth } = props;
 
-		const widthClasses = fullWidth
-			? `
-			mdc-layout-grid__cell
-			mdc-layout-grid__cell--span-12
-			`
-			: `
-			mdc-layout-grid__cell
-			mdc-layout-grid__cell--span-4-phone
-			mdc-layout-grid__cell--span-4-tablet
-			mdc-layout-grid__cell--span-6-desktop
-			`;
+	const widthClasses = fullWidth
+		? `
+		mdc-layout-grid__cell
+		mdc-layout-grid__cell--span-12
+		`
+		: `
+		mdc-layout-grid__cell
+		mdc-layout-grid__cell--span-4-phone
+		mdc-layout-grid__cell--span-4-tablet
+		mdc-layout-grid__cell--span-6-desktop
+		`;
 
-		/**
-		 * Filter whether to show a date range selector.
-		 *
-		 * Modules can op into the date range selection feature by returning true.
-		 */
-		const showDateRangeSelector = applyFilters( `googlesitekit.showDateRangeSelector-${ iconID }`, false );
+	// Determine whether the details cell should display.
+	const hasDetails = '' !== status || typeof props.children !== 'undefined';
```

Comment:
I can agree with that and will change this.

I will say that setting children directly as `null` or `false` seems like an anti-pattern unless you're intentionally wanting to bypass an unset prop conditional. I like keeping `null` and `undefined` values distinct as they have different meanings IMO. Checking an `undefined` type for `children` directly allows you to test components in isolation (no children rendered, to assert on wrapper elements) via `<Component children={ null } />` vs supplying a dummy component `<Component children={ () => {} } />`. I don't feel too strongly about it either way.

Changing this to the following to keep the return type of `hasDetails` a boolean. Without, it will have the return type of `false | React.Node`. A bit nit picky as `React.Node` is truthy, but may fail if someone later expects a boolean and operates on the value with a method that does not use type coercion or has unexpected coercion (handles React components in a special way). 

```js
const hasDetails = '' !== status || Boolean( children ); // returns true | false
```